nvdaHelper/vbufBackends/gecko_ia2/gecko_ia2.cpp

```diff
@@ -47,43 +48,73 @@ bool hasXmlRoleAttribContainingValue(const map<wstring,wstring>& attribsMap, con
 	return attribsMapIt != attribsMap.end() && attribsMapIt->second.find(roleName) != wstring::npos;
 }
 
-CComPtr<IAccessible2> GeckoVBufBackend_t::getRelationElement(
+std::vector<CComQIPtr<IAccessible2>> GeckoVBufBackend_t::getRelationElementsOfType(
 	LPCOLESTR ia2TargetRelation,
-	IAccessible2_2* element
+	IAccessible2_2* element,
+	const std::optional<std::size_t> numRelations
 ) {
-	IUnknown** ppUnk=nullptr;
-	long nTargets=0;
-	// We only need to request one relation target
-	int numRelations=1;
-	// However, a bug in Chrome causes a buffer overrun if numRelations is less than the total number of targets the node has.
-	// Therefore, If this is Chrome, request all targets (by setting numRelations to 0) as this works around the bug.
-	// There is no major performance hit to fetch all targets in Chrome as Chrome is already fetching all targets either way.
-	// In Firefox there would be extra cross-proc calls.
-	if(this->toolkitName.compare(L"Chrome")==0) {
-		numRelations=0;
+	constexpr long FETCH_ALL = 0l;  // See docs of relationTargetsOfType.
+	long nTargets = static_cast<long>(
+		min<size_t>(
+			numRelations.value_or(FETCH_ALL),  // Default is to fetch all relations.
+			static_cast<size_t>((numeric_limits<long>::max)())
+		)
+	);
+
+	const bool isChrome = this->toolkitName.compare(L"Chrome") == 0;
+	if (isChrome) {
+		// A bug in Chrome causes a buffer overrun if numRelations is less than the total number of targets the node has.
+		// TODO: has this been reported?

```

Comment:
is it worth testing, confirming and reporting this bug to Chromium? I don't think we have yet.

Reply:
It was reported to Chromium in https://bugs.chromium.org/p/chromium/issues/detail?id=1399184
However, the bug only states that Chromium ignores numRelations and gives back all available. It does not mention any buffer overrun. If there is in deed an overrun, it would be within Chromium, as chromium is the one that allocates the buffer.
Either way though, the general issue has been reported and Google will no doubt look into it.
Our work around is sufficient here.